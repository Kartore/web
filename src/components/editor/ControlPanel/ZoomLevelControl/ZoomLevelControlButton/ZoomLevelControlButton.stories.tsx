import type { Meta, StoryObj } from '@storybook/react';

import { ZoomLevelControlButton } from '.';

const meta = {
	component: ZoomLevelControlButton,
} satisfies Meta<typeof ZoomLevelControlButton>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
	args: {},
};
