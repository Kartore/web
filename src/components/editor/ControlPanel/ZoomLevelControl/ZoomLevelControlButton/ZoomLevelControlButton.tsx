import type { FC, ComponentPropsWithoutRef, MutableRefObject } from 'react';
import { useRef } from 'react';

import { type AriaButtonOptions, useButton } from 'react-aria';

import { cn } from '~/utils/tailwindUtil';

export type ZoomLevelControlButtonProps = {
  buttonRef?: MutableRefObject<HTMLButtonElement | null>;
} & AriaButtonOptions<'button'> &
  ComponentPropsWithoutRef<'button'>;

export const ZoomLevelControlButton: FC<ZoomLevelControlButtonProps> = ({
  className,
  ...props
}) => {
  const ref = useRef<HTMLButtonElement | null>(null);
  const { buttonRef = ref, ...otherProps } = props;
  const { buttonProps } = useButton(otherProps, buttonRef);

  return (
    <button
      {...props}
      ref={buttonRef}
      className={cn(
        'size-7 cursor-pointer rounded-lg border border-gray-300 bg-white p-1 transition-colors hover:bg-gray-100 active:bg-gray-300 disabled:cursor-default disabled:bg-white disabled:text-gray-300',
        className
      )}
      {...buttonProps}
    >
      {props.children}
    </button>
  );
};
