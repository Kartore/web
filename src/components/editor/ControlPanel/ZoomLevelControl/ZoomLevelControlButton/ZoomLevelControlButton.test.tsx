import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { describe } from 'vitest';

import * as Stories from './ZoomLevelControlButton.stories';

const { Default } = composeStories(Stories);

describe('Component: ZoomLevelControlButton', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
