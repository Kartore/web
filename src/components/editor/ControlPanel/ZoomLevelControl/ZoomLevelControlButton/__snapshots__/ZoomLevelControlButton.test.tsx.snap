// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: ZoomLevelControlButton > Snapshot > Default 1`] = `
<DocumentFragment>
  <button
    class="size-7 cursor-pointer rounded-lg border border-gray-300 bg-white p-1 transition-colors hover:bg-gray-100 active:bg-gray-300 disabled:cursor-default disabled:bg-white disabled:text-gray-300"
    type="button"
  />
</DocumentFragment>
`;
